### PR TITLE
feat(TMRS-226): transform text on euro widgets and add dynamic classes

### PR DESCRIPTION
**Pull request by douglasmik**
### Description

- Addition of transformation to test, e.g. `Quarter Finalist` to `QF`
- Addition of prop to allow us to add classes to widget components 

[TMRS-226](https://nidigitalsolutions.jira.com/browse/TMRS-226)

### Checklist

- [ ] Have you done any manual testing?
- [ ] Does it have automated tests?
- [ ] Do you need any other PRs merged before this (if so please list)?
- [ ] Do you need to update the README/Runbook
- [ ] Have you checked for [accessibility](https://nidigitalsolutions.jira.com/wiki/spaces/TNLDigital/pages/3898048523/Accessibility+Dev+Guide#Checklist)?

### Screenshots (if appropriate):

Include screenshots if needed.

[TMRS-226]: https://nidigitalsolutions.jira.com/browse/TMRS-226?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/ts-components/src/components/opta/football/fixtures-tournament/OptaFootballFixturesTournament.tsx [hunk 1]
```diff
@@ -24,7 +24,9 @@ export const OptaFootballFixturesTournament: React.FC<{
   days_before?: number;
   matchday?: string;
   round?: string;
+  classes?: string;
```
Comment (yukiinasawa) [inline packages/ts-components/src/components/opta/football/fixtures-tournament/OptaFootballFixturesTournament.tsx:27]: Minor: className might be more descriptive?

File: packages/ts-components/src/components/opta/football/player-stats/OptaFootballPlayerStats.tsx [hunk 2]
```diff
@@ -19,6 +19,7 @@ export const OptaFootballPlayerStats: React.FC<{
   season: string;
   competition: string;
   visible_categories: string; // goals | assists | cards_red | cards_yellow
+  classes?: string;
```
Comment (yukiinasawa) [inline packages/ts-components/src/components/opta/football/player-stats/OptaFootballPlayerStats.tsx:22]: Here as well

File: packages/ts-components/src/components/opta/football/standings/OptaFootballStandings.tsx [hunk 3]
```diff
@@ -19,6 +19,7 @@ export const OptaFootballStandings: React.FC<{
   season: string;
   competition: string;
   default_nav?: string;
+  classes?: string;
```
Comment (yukiinasawa) [inline packages/ts-components/src/components/opta/football/standings/OptaFootballStandings.tsx:22]: and here

**Review by yukiinasawa — APPROVED** [review]

File: packages/ts-components/src/components/opta/football/fixtures-tournament/OptaFootballFixturesTournament.tsx (hunk 1 above)
Comment (douglasmik) [reply packages/ts-components/src/components/opta/football/fixtures-tournament/OptaFootballFixturesTournament.tsx:27]: I normally do `classes` to differentiate from component `className`, where multiple 'classes' can be passed down

File: packages/ts-components/src/components/opta/utils/replaceNationalTeamDetails.ts [hunk 4]
```diff
@@ -18,11 +18,23 @@ export const replaceWithTBD = (element: HTMLCollectionOf<Element>) => {
...
-        country &&
-        (country.includes('Third Place') || country.includes('Group'))
-      ) {
-        (optaFlagContainer as HTMLElement).innerText = 'TBD';
+      if (country) {
+        if (country.includes('Third Place') || country.includes('Group')) {
+          (optaFlagContainer as HTMLElement).innerText = 'TBD';
+        }
+
+        (optaFlagContainer as HTMLElement).innerText = (optaFlagContainer as HTMLElement).innerText.replace(
```
Comment (HarisHodzicnt) [inline packages/ts-components/src/components/opta/utils/replaceNationalTeamDetails.ts:26]: optinal, we can write this as:

const optaFlagContainer = document.getElementById('optaFlagContainer') as HTMLElement;

if (optaFlagContainer) {
    const replacements = [
        { pattern: /Quarter-Finalist/g, replacement: 'QF' },
        { pattern: /Semi-Finalist/g, replacement: 'SF' },
        { pattern: /Semi-Final/g, replacement: 'SF' }
    ];

    replacements.forEach(({ pattern, replacement }) => {
        optaFlagContainer.innerText = optaFlagContainer.innerText.replace(pattern, replacement);
    });
}

**Review by HarisHodzicnt — APPROVED** [review]

File: packages/ts-components/src/components/opta/utils/replaceNationalTeamDetails.ts (hunk 4 above)
Comment (douglasmik) [reply packages/ts-components/src/components/opta/utils/replaceNationalTeamDetails.ts:26]: A bit different but agree with the approach - this is more flexible, will update  👍

**Pull request merged**